engine/api/vcs/dao_vcs_project.go

```diff
@@ -0,0 +1,86 @@
+package vcs
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-gorp/gorp"
+	"github.com/ovh/cds/engine/api/database/gorpmapping"
+	"github.com/ovh/cds/engine/gorpmapper"
+	"github.com/ovh/cds/sdk"
+	"github.com/rockbears/log"
+)
+
+func Insert(ctx context.Context, db gorpmapper.SqlExecutorWithTx, vcsProject *sdk.VCSProject) error {
+	vcsProject.ID = sdk.UUID()
+	vcsProject.Created = time.Now()
+	dbData := &dbVCSProject{VCSProject: *vcsProject}
+	if err := gorpmapping.InsertAndSign(ctx, db, dbData); err != nil {
+		return err
+	}
+	*vcsProject = dbData.VCSProject
+	return nil
+}
+
+func Update(ctx context.Context, db gorpmapper.SqlExecutorWithTx, vcsProject *sdk.VCSProject) error {
+	var dbData = dbVCSProject{VCSProject: *vcsProject}
+	if err := gorpmapping.UpdateAndSign(ctx, db, &dbData); err != nil {
+		return err
+	}
+	*vcsProject = dbData.VCSProject
+	return nil
+}
+
+func Delete(db gorpmapper.SqlExecutorWithTx, projectID int64, name string) error {
+	_, err := db.Exec("DELETE FROM vcs_project WHERE project_id = $1 AND name = $2", projectID, name)
+	return sdk.WrapError(err, "cannot delete vcs_project %d/%s", projectID, name)
+}
+
+func LoadAllVCSByProject(ctx context.Context, db gorp.SqlExecutor, projectID int64) ([]sdk.VCSProject, error) {
+	return loadAllVCSByProject(ctx, db, projectID)
```

Comment:
Is this func needed ?

Reply:
yes, I think: to list the vcs configured on a project